source/extensions/filters/http/wasm/wasm_filter.cc

```diff
@@ -23,7 +149,7 @@ FilterConfig::FilterConfig(const envoy::extensions::filters::http::wasm::v3::Was
     });
   };
 
-  if (!Common::Wasm::createWasm(plugin, context.scope().createScope(""), server.clusterManager(),
+  if (!Common::Wasm::createWasm(plugin, context.scope().createScope(""), mycluster /*server.clusterManager()*/,
```

Comment:
here is the main issue I found. server.clusterManager() assert due to the manager is still not initialized (I guess) at this moment

Reply:
Where is this assert failing?  was it a manual run of Envoy with upstream filter?  If it's in a unit test, the unit test is probably just not set up correctly.  if you add an e2e test with wasm upstream filters it will give me something Ican patch and run.

Usually I suggest folks doing upstream filter work parameterize the e2e test using prependFilter like we did for test/integration/filter_integration_test.cc

Reply:
Yeah, that's the same issue: https://github.com/envoyproxy/envoy/issues/33218, which is resolved now.